src/plots/plots.js

```diff
@@ -1095,57 +1090,11 @@ plots.supplyDataDefaults = function(dataIn, dataOut, layout, fullLayout) {
         fullTrace._input = trace;
         fullTrace._expandedIndex = cnt;
 
-        if(fullTrace.transforms && fullTrace.transforms.length) {
-            var sdInvisible = trace.visible !== false && fullTrace.visible === false;
-
-            var expandedTraces = applyTransforms(fullTrace, dataOut, layout, fullLayout);
-
-            for(var j = 0; j < expandedTraces.length; j++) {
-                var expandedTrace = expandedTraces[j];
-
-                // No further templating during transforms.
-                var fullExpandedTrace = {
-                    _template: fullTrace._template,
-                    type: fullTrace.type,
-                    // set uid using parent uid and expanded index
-                    // to promote consistency between update calls
-                    uid: fullTrace.uid + j
-                };
-
-                // If the first supplyDefaults created `visible: false`,
-                // clear it before running supplyDefaults a second time,
-                // because sometimes there are items we still want to coerce
-                // inside trace modules before determining that the trace is
-                // again `visible: false`, for example partial visibilities
-                // in `splom` traces.
-                if(sdInvisible && expandedTrace.visible === false) {
-                    delete expandedTrace.visible;
-                }
-
-                plots.supplyTraceDefaults(expandedTrace, fullExpandedTrace, cnt, fullLayout, i);
-
-                // relink private (i.e. underscore) keys expanded trace to full expanded trace so
-                // that transform supply-default methods can set _ keys for future use.
-                relinkPrivateKeys(fullExpandedTrace, expandedTrace);
+        // add identify refs for consistency with transformed traces
+        fullTrace._fullInput = fullTrace;
+        fullTrace._expandedInput = fullTrace;
```

Comment:
Without transforms there will always be a 1:1 relationship between `gd.data` and `gd._fullData` entries, right? So we should be able to entirely get rid of `_expandedIndex` (just use `index`) as well as `_fullInput` and `_expandedInput` (just use `_input`)

Reply:
Good call. Addressed in [89e2321](https://github.com/plotly/plotly.js/pull/7240/commits/89e232177c33696add5e711cc0e09043d4d20ebe).

@alexcjohnson If I understand correctly, you want me to remove `gd._fullData` and use `gd.data` everywhere instead?

Reply:
> Good call. Addressed in [89e2321](https://github.com/plotly/plotly.js/pull/7240/commits/89e232177c33696add5e711cc0e09043d4d20ebe).
> 
> @alexcjohnson If I understand correctly, you want me to remove `gd._fullData` and use `gd.data` everywhere instead?

That won't work. So the current state of the PR should be good enough.

Reply:
Ah sorry, got the substitution wrong. Looks like `_expandedInput` is already gone, but now since `fullTrace._fullInput` is always just `fullTrace`, shouldn't it be possible to remove it everywhere? ie when you have a line like
```js
var iIn = inputIndices[i] = trace._fullInput.index;
```
it can turn into
```js
var iIn = inputIndices[i] = trace.index;
```
And then there will be further simplifications because what you're creating is often now the same as something else, or the whole block is unnecessary like:

https://github.com/plotly/plotly.js/blob/a5b202c859faaa3362ba0a2e9e5ea31b001bdaaa/src/components/selections/select.js#L970-L972

because that would become `if (trace !== trace)`

Reply:
Addressed in [b22f62b](https://github.com/plotly/plotly.js/pull/7240/commits/b22f62b5c0f29906ccef6618fb539967c151e7c9).

Reply:
Again, you should be able to remove `_fullInput` entirely at this point - not just in the cases where you're accessing `_fullInput.index`. This isn't causing any problems, it's just cruft we should remove at some point. If you want to move on and make an issue to come back to this later then sure, but here's what I see at the current state of this PR:

```js
 ~/plotly/plotly.js/src > drop-transforms > ag _fullInput
traces/parcoords/plot.js
60:        var preGUI = fullLayout._tracePreGUI[gd._fullData[fullIndices[i]]._fullInput.uid];
119:        //     fullLayout._tracePreGUI[gd._fullData[fullIndices[i]]._fullInput.uid],

traces/table/data_preparation_helper.js
41:    var columnOrder = trace._fullInput.columnorder;

components/legend/handle_click.js
47:        fullTrace = fullTrace._fullInput;
89:        var fullInput = fullTrace._fullInput || fullTrace;
108:    var fullInput = fullTrace._fullInput;

components/legend/draw.js
113:                _fullInput: shape,
559:                var fullInput = legendItem.trace._fullInput || {};

components/selections/select.js
946:        var fullInputTrace = searchTraces[i].cd[0].trace._fullInput;
958:            trace._input.selectedpoints = trace._fullInput.selectedpoints = [];
959:            if(trace._fullInput !== trace) trace.selectedpoints = [];
970:                if(trace._fullInput !== trace) {
975:                if(trace._fullInput !== trace) {
985:            if(trace._fullInput !== trace) {
986:                delete trace._fullInput.selectedpoints;

plot_api/template_api.js
345:            if(!fullTrace._fullInput._template) {

plot_api/plot_api.js
1445:            var preGUI = fullLayout._tracePreGUI[getFullTrace(tracei)._fullInput.uid];
1517:            var preGUI = fullLayout._tracePreGUI[contFull._fullInput.uid];
2400:        if(fullData[i]._fullInput.uid === uid) return i;
2521:                fullInput = fullTrace._fullInput;
2784:            trace = newFullData[i]._fullInput;
2788:            getDiffFlags(oldFullData[i]._fullInput, trace, [], diffOpts);

plots/cartesian/type_defaults.js
121:    var isCandlestick = traceIs(trace._fullInput || {}, 'candlestick');

plots/plots.js
502:        uid = newFullData[i]._fullInput.uid;
554:        var thisFullInput = oldFullData[i]._fullInput;
1091:        fullTrace._fullInput = fullTrace;
```